src/java.desktop/macosx/classes/sun/lwawt/macosx/CPrinterJob.java

```diff
@@ -788,7 +788,18 @@ private void printToPathGraphics(    final PeekGraphics graphics, // Always an a
                 Graphics2D pathGraphics = new CPrinterGraphics(delegate, printerJob); // Just stores delegate into an ivar
                 Rectangle2D pageFormatArea = getPageFormatArea(page);
                 initPrinterGraphics(pathGraphics, pageFormatArea);
-                painter.print(pathGraphics, FlipPageFormat.getOriginal(page), pageIndex);
+                if (monochrome) {
+                    BufferedImage bufferedImage = new BufferedImage((int)page.getWidth(), (int)page.getHeight(),
+                            BufferedImage.TYPE_INT_ARGB);
+                    Graphics2D g2 = bufferedImage.createGraphics();
```

Comment:
The problem here is that the image is not at printer resolution.
So it will be VERY grainy when printed.
If you are doing this band printing you need to do something like what is happening in RasterPrinterJob.
In fact probably defer to that.
And this is definitely a "mis-use" of path graphics.
On top of that I think it odd to create an image of BufferedImage.TYPE_INT_ARGB and then convert it to greyscale. Why not just create a BI of type BufferedImage.TYPE_USHORT_GRAY ? And then the conversion is done by 2D when drawing into it and there's no need to filter the image.
I tested that but for whatever reason I needed to clear the image to white first.
But I don't think you should be "here" if the only way you can do monochrome on mac is by printing a greyscale image. It should go to the existing code in RasterPrinterJob which will handle the resolution. It would be wrong to replicate that code here. I am not sure that code is used today in the mac printing path,. but it is the code that does what you need .. mostly.